### DISCUSSION TITLE
Evolution of the interface

### DISCUSSION THREAD

**Discussion by Jean-Gabriel1982** [GrandOrgue/grandorgue#999] [General]
For a few months, the development of GrandOrgue has taken a new impetus, by adding new things, and fixing bugs.

Perhaps it would also be the moment to think of a new graphic interface?
Maybe for a 4.0 or 5.0 version of GrandOrgue?

I think it would be a strong signal to the Community of organists and enthusiasts that a substantial modernization of the interface of GrandOrgue be carried out.
I think @Mikelectric would agree with this idea ;)

Maybe, in the same way as for the icon, suggestions would be welcome?
@Mikelectric, maybe you can survey your forum, there may be very good graphic designers, and happy to help ?

IMHO, it is necessary to think the interface towards the tactile and not the use of the mouse.

**Comment by matheus-j-ressurreicao** [comment]
I really liked the idea. If any good graphic designers are willing to help, I'm sure it will be of great benefit. I also think it would be very useful to discuss what would really be interesting to appear in this new graphical interface, thinking about issues such as the presence of this or that icon contributing to the usability of the program as a whole. For example, if I am not mistaken, in GrandOrgue's current control panel there is a bar, very similar to the volume bar, which shows, I believe, the overload of some computational resource (I suppose the CPU). I am not sure how it works and what it contributes to the use of the program, so I think this update will be very useful.

**Comment by willeke1234** [comment]
Is a new graphic interface a new look with better graphics or is it a less confusing, more consistent and more easy to use GUI?

**Comment by Jean-Gabriel1982** [comment]
Hello all,

One idea for modernise GO interface ... one test light and dark mode ...
if you like the concept, I could continue some ideas with different menus for example...

![interface_go_light](https://user-images.githubusercontent.com/20370038/154641979-9e498762-4029-4703-9961-89c68b51c712.jpg)
![interface_go_dark](https://user-images.githubusercontent.com/20370038/154641965-9c0437e6-3495-4290-9eb1-ec06766c5b9b.jpg)

**Comment by oleg68** [comment]
@Jean-Gabriel1982 Thank you for your ideas.

You propose the screen with organ icons. But there are some questions:

1. Should this screen be always visible or only when there is no organ loaded?
2. Most of organs does not have an icon. How to choose them on the screen?
3. Now maintaining the lists of organs is in the `Organs` tab of the `Settings` dialog. Do you propose to get this tab rid in favor to the new screen?

**Reply by Jean-Gabriel1982** [reply]
@oleg68 

> @Jean-Gabriel1982 Thank you for your ideas.
> 
> 
> 
> You propose the screen with organ icons. But there are some questions:
> 
> 
> 
> 1. Should this screen be always visible or only when there is no organ loaded?
> 

in fact, the principle is to always keep visible (except when the organ is in full screen) the upper bar (incomplete in the example that I proposed). 

thus, the different "screens" (parameters, organ list etc) appear when they are "called up" via the top bar.

In my example, this screen should appear when the user click on "load". 

> 2. Most of organs does not have an icon. How to choose them on the screen?
> 
I think when an organ is loaded, the user can choose him/herself the icon he/she wants to be the icon. 


> 3. Now maintaining the lists of organs is in the `Organs` tab of the `Settings` dialog. Do you propose to get this tab rid in favor to the new screen?

Yes and no.  I haven't considered all the options yet. 

I think it would be interesting for the GO interface to evolve towards a tactile interface, in order to facilitate the interaction between the user and his screen, while keeping this possibility of using the mouse if the user wishes. 

in the case of your question, we could imagine a short press on the icon, which loads the organ, and a long press (at least 3 seconds) which opens the properties, and even possibly allows you to rearrange the list order of the organs...
